test/cache-interceptor/cache-stores.js

```diff
@@ -0,0 +1,167 @@
+const { describe, test } = require('node:test')
+const { deepStrictEqual, strictEqual } = require('node:assert')
+const MemoryCacheStore = require('../../lib/cache/memory-cache-store')
+
+const cacheStoresToTest = [
+  MemoryCacheStore
+]
+
+for (const CacheStore of cacheStoresToTest) {
```

Comment:
We should make this an abstract test which gets reused.

Something that can get used like https://github.com/fastify/fastify/blob/main/test/put.test.js